embedded-graphics/src/draw_target.rs

```diff
@@ -0,0 +1,304 @@
+use crate::{
+    drawable::{self, Drawable},
+    geometry::{Point, Size},
+    pixelcolor::PixelColor,
+    primitives::{self, Primitive},
+    style::{PrimitiveStyle, Styled},
+};
+
+/// Defines a display that can be used to render [`Drawable`] objects.
+///
+/// To use this crate in a driver, `DrawTarget` must be implemented. This trait defines how a
+/// display draws pixels, and optionally provides a way to define accelerated drawing methods for
+/// graphical primitives such as lines, rectangles, triangles, and circles.
+///
+/// Once a `DrawTarget` is defined, it can be used to render [`Drawable`]s. Note that any iterator
+/// over [`Pixel`]s has a default implementation for the [`Drawable`] trait. See the [`Drawable`]
+/// trait documentation for more details.
+///
+/// Here's an example for an imaginary display that has a 64x64px framebuffer of 8 bit values that
+/// communicates over a (simplified) SPI interface:
+///
+/// ```rust
+/// use embedded_graphics::{
+///     drawable::Pixel,
+///     egcircle,
+///     geometry::Size,
+///     pixelcolor::{Gray8, GrayColor},
+///     prelude::*,
+///     primitive_style, DrawTarget,
+/// };
+///
+/// # struct SPI1;
+/// #
+/// # impl SPI1 {
+/// #     pub fn send_bytes(&self, buf: &[u8]) -> Result<(), ()> {
+/// #         Ok(())
+/// #     }
+/// # }
+/// #
+/// /// A fake display 64px x 64px where each pixel is stored as a single `u8`
+/// struct ExampleDisplay {
+///     framebuffer: [u8; 64 * 64],
+///     iface: SPI1,
+/// }
+///
+/// impl ExampleDisplay {
+///     /// Send buffer to the display
+///     pub fn flush(&self) -> Result<(), ()> {
+///         self.iface.send_bytes(&self.framebuffer)
+///     }
+/// }
+///
+/// impl DrawTarget<Gray8> for ExampleDisplay {
+///     type Error = core::convert::Infallible;
+///
+///     /// Draw a `Pixel` that has a color defined as `Gray8`.
+///     fn draw_pixel(&mut self, pixel: Pixel<Gray8>) -> Result<(), Self::Error> {
```

Comment:
Doesn't handle negative coordinates correctly. But fixing this could wait until `TryFrom<Point> for (u32, u32)` is merged.

Reply:
Yeah let's wait for that to get merged. Much more elegant solution. Should probably get this PR rebased on a merged #245 before merging into master?

Reply:
Not sure when I have time to review #245. So if you want to merge this PR first just make these changes in #245.

Reply:
Sounds good to me :+1: